sky/skylet/subprocess_daemon.py

```diff
@@ -28,29 +46,33 @@
     if process is None:
         sys.exit()
 
+    children = []
     if parent_process is not None:
         # Wait for either parent or target process to exit.
         while process.is_running() and parent_process.is_running():
+            try:
+                # if process is terminated by the time reaching this line,
+                # it returns an empty list.
+                tmp_children = process.children(recursive=True)
+                if tmp_children:
+                    children = tmp_children
+                    children.append(process)
+            except psutil.NoSuchProcess:
+                pass
```

Comment:
Why do we move this in the while loop? Is it to make sure that even if the `process` is killed, we have recorded all the child processes before the process is killed, so we can kill those processes to avoid leakage? 

If that is the purpose, we should mention that in the comment. (style nit: comment should be more about `why` we do a thing, instead of `what` the code does).

Reply:
The reason you mentioned is exactly correct on why I moved this into `while `loop. This was a second issue of this script. Running `process.children(recursive=True)` after the process got killed returned empty list so it was not able to remove the child processes although it was daemonized.

Thanks for the reminder advice! The comment is updated at [4e1e7f3](https://github.com/skypilot-org/skypilot/pull/3919/commits/4e1e7f3ea6ff0a1d96fd11ee3309623583572193)